akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -544,15 +548,15 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
     val (ctrl, completed) =
       if (remoteSettings.TestMode) {
```

Comment:
@drewhk Here's another thing I wanted to ask you. As you can see here we materialize in two different ways depending on if TestMode is enabled. Additional materialized value. Do you think it would be alright to always include the TestStage, but have it act as plain pass-through when TestMode is disabled. Performance overhead?